### PR TITLE
Fix bug where cluster in deletion kills controller

### PR DESCRIPTION
**Pull request by phiphi282**
**What this PR does / why we need it**:
Sometimes a cluster in deletion can kill the controller possibly
indefinitely, when it tries to add a finalizer to them.
This happens for example when a cluster is in deletion and the
SecretsCleanupFinalizer gets removed when the controller-manager
restarts for some reason.
Then it will always try to add this finalizer again and fail.

Make sure that finalizers only get added when objects are not stuck in
deletion.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: api/pkg/crd/migrations/seed/migrations.go
Comment (xrstf) [inline api/pkg/crd/migrations/seed/migrations.go]: If I read this correctly, this will not add the finalizer when the cluster is already in deletion. That make sense.

But in line 228, right before this, we create the credentials Secrets regardless of the deletion status. Won't these secrets then be orphans once the cluster is gone?

**Review by xrstf — COMMENTED** [review]
Some questions, otherwise nice catch :-)

File: api/pkg/crd/migrations/seed/migrations.go
Comment (phiphi282) [reply api/pkg/crd/migrations/seed/migrations.go]: Ah yes, changed it to just return at the beginning of the function if there is a deletionTimestamp.
Make more sense it is even better readable this way :D

**Review by xrstf — APPROVED** [review]
/approve
/hold

Holding until the release-blocking PRs are merged.

**Pull request merged**
